crates/router/src/core/payments.rs

```diff
@@ -319,7 +319,7 @@ impl PaymentRedirectFlow for PaymentRedirectCompleteAuthorize {
             metadata: Some(Metadata {
                 order_details: None,
                 data: masking::Secret::new("{}".into()),
-                payload: Some(req.json_payload.unwrap_or("{}".into()).into()),
+                payload: Some(req.json_payload.unwrap_or_else(|| "{}".into()).into()),
```

Comment:
Correct me If I am wrong, here:
- json_payload: `Option<Value>`
- payload: `Option<SecretSerdeValue>`,

By doing this operation that is performed here, the value that is being yielded when json_payload is `None` will be `serde_json::Value::String("{}")`, But shouldn't this be instead `serde_json::Value::Object {}`

Reply:
```suggestion
                payload: Some(req.json_payload.unwrap_or(serde_json::json!({}))).into()),
```